src/Tools/dotnet-stack/ReportCommand.cs

```diff
@@ -55,91 +77,125 @@ private static async Task<int> Report(CancellationToken ct, TextWriter stdOutput
                     stdError.WriteLine("Process ID should not be negative.");
                     return -1;
                 }
-                else if (processId == 0)
+
+                DiagnosticsClientBuilder builder = new("dotnet-stack", 10);
+                using (DiagnosticsClientHolder holder = await builder.Build(ct, processId, diagnosticPort, showChildIO: false, printLaunchCommand: false).ConfigureAwait(false))
                 {
-                    stdError.WriteLine("--process-id is required");
-                    return -1;
-                }
+                    if (holder == null)
+                    {
+                        return -1;
+                    }
 
+                    DiagnosticsClient client = holder.Client;
 
-                DiagnosticsClient client = new(processId);
-                List<EventPipeProvider> providers = new()
-                {
-                    new EventPipeProvider("Microsoft-DotNETCore-SampleProfiler", EventLevel.Informational)
-                };
-
-                // collect a *short* trace with stack samples
-                // the hidden '--duration' flag can increase the time of this trace in case 10ms
-                // is too short in a given environment, e.g., resource constrained systems
-                // N.B. - This trace INCLUDES rundown.  For sufficiently large applications, it may take non-trivial time to collect
-                //        the symbol data in rundown.
-                EventPipeSession session = await client.StartEventPipeSessionAsync(providers, requestRundown:true, token:ct).ConfigureAwait(false);
-                using (session)
-                using (FileStream fs = File.OpenWrite(tempNetTraceFilename))
-                {
-                    Task copyTask = session.EventStream.CopyToAsync(fs, ct);
-                    await Task.Delay(duration, ct).ConfigureAwait(false);
-                    await session.StopAsync(ct).ConfigureAwait(false);
-
-                    // check if rundown is taking more than 5 seconds and add comment to report
-                    Task timeoutTask = Task.Delay(TimeSpan.FromSeconds(5));
-                    Task completedTask = await Task.WhenAny(copyTask, timeoutTask).ConfigureAwait(false);
-                    if (completedTask == timeoutTask)
+                    // Resume runtime if it was suspended (similar to --resume-runtime:true in other tools)
+                    // This is safe to call even if the runtime wasn't suspended - it's a no-op in that case
+                    try
                     {
-                        stdOutput.WriteLine($"# Sufficiently large applications can cause this reportCommand to take non-trivial amounts of time");
+                        await client.ResumeRuntimeAsync(ct).ConfigureAwait(false);
+                    }
+                    catch (Exception)
```

Comment:
Its a little confusing that if it we deemed it safe to call ResumeRuntimeAsync, that we would expect exceptions. If the runtime was suspended and we hit an exception, we are now ignoring it, is that right?